### PR TITLE
adding custom errors for elastic.io errors

### PR DESCRIPTION
**Pull request by Menshawi**

### CODE REVIEW

File: codes.json [hunk 1]
```diff
@@ -868,6 +868,54 @@
...
+  "MethodNotAllowed": {
+    "status": 405,
+    "code": 4050093,
+    "error": "method_not_allowed",
+    "description": {
+      "en": "Invalid method provided."
+    }
+  },
+  "InvalidFormat": {
+    "status": 406,
```
Comment (XinV) [inline codes.json]: Our invalid Param usually use 400, any reason you use 406?

File: codes.json [hunk 2]
```diff
@@ -868,6 +868,54 @@
...
+  "UnauthorizedApiAccess": {
+    "status": 401,
+    "code": 4010092,
+    "error": "unauthorized_api_access",
+    "description": {
+      "en": "Access is denied, please check your API key."
+    }
+  },
+  "MethodNotAllowed": {
+    "status": 405,
```
Comment (XinV) [inline codes.json]: Why 405?

File: codes.json [hunk 3]
```diff
@@ -868,6 +868,54 @@
...
+  "InvalidFormat": {
+    "status": 406,
+    "code": 4060094,
+    "error": "invalid_format",
+    "description": {
+      "en": "You requested a format that isn't JSON."
+    }
+  },
+  "TooManyRequests": {
+    "status": 429,
```
Comment (XinV) [inline codes.json]: Why 429?

File: codes.json [hunk 4]
```diff
@@ -868,6 +868,54 @@
...
+    "code": 4290095,
+    "error": "too_many_requests",
+    "description": {
+      "en": "You're performing too many requests."
+    }
+  },
+  "IntegrationServiceDown": {
+    "status": 503,
```
Comment (XinV) [inline codes.json]: Can you check we use 503 for service down? Is it using 503 in practice

Comment (Menshawi) [inline codes.json]: we don't have any error defined for service down, this is on Elastic.io side

File: codes.json (hunk 3 above)
Comment (Menshawi) [inline codes.json]: these are all status codes defined by Elastic.io

Comment (Menshawi) [inline codes.json]: https://api.elastic.io/docs/#errors

File: codes.json (hunk 1 above)
Comment (Menshawi) [inline codes.json]: these are the status codes used by Elastic.io

File: codes.json (hunk 2 above)
Comment (Menshawi) [inline codes.json]: same reason as above

**Pull request merged**